platforms/software/build-init/src/main/java/org/gradle/buildinit/tasks/InitBuild.java

```diff
@@ -485,33 +458,33 @@ private GradleException createNotSupportedTestFrameWorkException(BuildInitialize
 
     @VisibleForTesting
     String getEffectiveProjectName(UserQuestions userQuestions, BuildInitializer initializer) {
-        String projectName = this.projectName;
         if (initializer.supportsProjectName()) {
-            if (isNullOrEmpty(projectName)) {
-                return userQuestions.askQuestion("Project name", getProjectName());
+            if (getProjectName().isPresent()) {
```

Comment:
🤔 I wonder if you technically can have project name present and it's an empty string?

Could that be a problem? Do we need to check:
```if (isNullOrEmpty(getProjectName().getOrNull()) {}```
?

Reply:
I think it is possible to set it to "", but in that case, this means the user specifically specified that value, and there is no reason to go into interactive mode.